panic: assignment to entry in nil map

panic: assignment to entry in nil map
 
goroutine 9 [running]:
panic(0x569c20, 0xc82046a390)
        /usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
main.(*Handler).EndBatch(0xc820064120, 0xc82044e800, 0x0, 0x0)
        /home/administrator/go/src/github.com/nathanielc/morgoth/cmd/morgoth/main.go:340 +0x356
github.com/nathanielc/morgoth/vendor/github.com/influxdata/kapacitor/udf/agent.(*Agent).readLoop(0xc820052150, 0x0, 0x0)
        /home/administrator/go/src/github.com/nathanielc/morgoth/vendor/github.com/influxdata/kapacitor/udf/agent/agent.go:209 +0x4fe
github.com/nathanielc/morgoth/vendor/github.com/influxdata/kapacitor/udf/agent.(*Agent).Start.func1(0xc820052150)
        /home/administrator/go/src/github.com/nathanielc/morgoth/vendor/github.com/influxdata/kapacitor/udf/agent/agent.go:89 +0x6b
created by github.com/nathanielc/morgoth/vendor/github.com/influxdata/kapacitor/udf/agent.(*Agent).Start
        /home/administrator/go/src/github.com/nathanielc/morgoth/vendor/github.com/influxdata/kapacitor/udf/agent/agent.go:98 +0x1bb
 
------
ID: dev-morgoth
Error:
Template:
Type: stream
Status: enabled
Executing: true
Created: 29 Jun 17 22:44 UTC
Modified: 10 Jul 17 23:33 UTC
LastEnabled: 10 Jul 17 23:33 UTC
Databases Retention Policies: ["stats"."autogen"]
TICKscript:
stream
    |from()
        .measurement('traf')
        .groupBy(*)
    |window()
        .periodCount(60)
        .everyCount(60)
    @morgoth()
        .field('traffic_out')
        .scoreField('anomalyScore')
        .sigma(3.5)
    |alert()
        .details('error')
        .crit(lambda: "anomalyScore" > 0.9)
        .log('/tmp/dev-morgoth-alerts.log')

--- a/cmd/morgoth/main.go
+++ b/cmd/morgoth/main.go
@@ -308,7 +308,7 @@ func (h *Handler) Point(p *agent.Point) error {
 		if i, ok := p.FieldsInt[h.field]; ok {
 			value = float64(i)
 		} else {
-			return fmt.Errorf("no field %s is not a float or int", h.field)
+			return fmt.Errorf("field %q is not a float or int", h.field)
 		}
 	}
 	h.currentWindow.Data = append(h.currentWindow.Data, value)
@@ -337,6 +337,9 @@ func (h *Handler) EndBatch(b *agent.EndBatch) error {
 		}
 		for _, p := range h.batchPoints {
 			if h.scoreField != "" {
+				if p.FieldsDouble == nil {
+					p.FieldsDouble = make(map[string]float64, 1)
+				}
 				p.FieldsDouble[h.scoreField] = 1 - avgSupport
 			}
 			h.agent.Responses <- &agent.Response{

